release-content/migration-guides/reflect_registration_changes.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Changes to type registration for reflection
+pull_requests: [15030, 20435, 20893]
+---
+
+Calling `.register_type` has long been a nuisance for Bevy users: both library authors and end users.
+This step was previously required in order to register reflected type information in the `TypeRegistry`.
+
+This comes with a few caveats however:
+
+1. This functionality is gated by feature flags. Projects with default features off will not see reflected types unless one of these features is enabled.
+2. There are two approaches to do this: one has incomplete platform support, while the other relies on a specific project structure.
+3. Generic types are not automatically registered, and must still be manually registered.
+
+In order for Bevy to automatically register your types, you need to turn on the `reflect_auto_register` feature, or the fallback `reflect_autoregister_static`.
+The `reflect_auto_register` feature is part of Bevy's default features, and is mutually incompatible with `reflect_autoregister_static`.
+Be aware that the `reflect_autoregister_static` feature comes with some caveats for project structure: check the docs for [load_type_registrations!](https://docs.rs/bevy/0.17.0-rc.1/bevy/reflect/macro.load_type_registrations.html).
+
+We recommend:
+
+1. Enable `reflect_auto_register` in your application code, CI and in examples/tests. You can enable `bevy` features for tests only by adding a matching copy of `bevy` to `dev-dependencies` with the needed features enabled.
+   1. Many libraries, and most production games do not need this functionality. This feature flag (like all reflection) is often only used for development, not for production projects.
```

Comment:
Disagree about "most production games" not needing it. Skein and LDtk, which are pretty popular, both use reflection in production. TrenchBroom doesn't *require* is theoretically, but heavily encourages you to ship with it enabled to allow for community map modding. BEI was toying with the idea of basing their workflow around auto reflection. I believe giving people this tip right now could give them the wrong idea about how the ecosystem uses reflection.

Reply:
I've amended this: please let me know if you agree with the new wording!

Reply:
Sounds good to me :)